docs/develop/typescript/ai-sdk.mdx

```diff
@@ -0,0 +1,126 @@
+---
+id: ai-sdk
+title: AI SDK - TypeScript SDK
+sidebar_label: AI SDK
+toc_max_heading_level: 2
+keywords:
+  - ai
+  - agents
+tags:
+  - AI SDK
+  - TypeScript SDK
+  - Temporal SDKs
+description: Implement Agentic applications in TypeScript using the Temporal TypeScript SDK and the AI SDK.
+---
+
+Temporal's integration with the [AI SDK](https://ai-sdk.dev/) in Typescript brings Temporal's durability to the easy development of agentic applications provided by the AI SDK.
+
+## How to configure to use the AI SDK
+
+The TypeScript SDK comes with an optional ai-sdk package that provides a plugin which configures workers to use the AI SDK.
+See how to use it in the [ai-sdk](https://github.com/temporalio/samples-typescript/tree/main/ai-sdk/src/worker.ts) code sample.
+
+`modelProvider` is a `ProviderV2` which will be used to create models.
+
+```ts
+import { openai } from '@ai-sdk/openai';
+
+const worker = await Worker.create({
+      plugins: [
+        new AiSDKPlugin({
```

Comment:
What does this do exactly? If we are already specifying the model in the workflow code, what's the benefit of doing this to workers?